rotkehlchen/chain/evm/decoding/uniswap/v4/utils.py

```diff
@@ -100,3 +113,128 @@ def get_uniswap_v4_position_price(
         tick=tick,
         price_func=price_func,
     )
+
+
+def decode_uniswap_v4_like_swaps(
+        transaction: 'EvmTransaction',
+        decoded_events: list['EvmEvent'],
+        all_logs: list['EvmTxReceiptLog'],
+        base_tools: 'BaseDecoderTools',
+        swap_topics: tuple[bytes, ...],
+        counterparty: str,
+        router_address: ChecksumEvmAddress,
+        wrapped_native_currency: 'Asset',
+) -> list['EvmEvent']:
+    """Decode Uniswap V4 like swaps."""
+    amounts_received, amounts_sent, pools_used, possible_fees = set(), set(), set(), defaultdict(list)  # noqa: E501
```

Comment:
use a set since what you do is to check if you have a fee amount in there


```suggestion
    amounts_received, amounts_sent, pools_used, possible_fees = set(), set(), set(), defaultdict(set)  # noqa: E501
```